src/panels/lovelace/cards/hui-gauge-card.ts

```diff
@@ -1,15 +1,127 @@
-import { html } from "@polymer/polymer/lib/utils/html-tag.js";
-import { PolymerElement } from "@polymer/polymer/polymer-element.js";
+import { html, LitElement, PropertyDeclarations } from "@polymer/lit-element";
+import { LovelaceCard, LovelaceConfig } from "../types.js";
+import { HomeAssistant } from "../../../types.js";
+import isValidEntityId from "../../../common/entity/valid_entity_id.js";
+import { fireEvent } from "../../../common/dom/fire_event.js";
 
 import "../../../components/ha-card.js";
 
-import EventsMixin from "../../../mixins/events-mixin.js";
+interface Config extends LovelaceConfig {
+  entity: string;
+  title?: string;
+  unit_of_measurement?: string;
+  min?: number;
+  max?: number;
+  severity?: object;
+}
+
+class HuiGaugeCard extends LitElement implements LovelaceCard {
+  public hass?: HomeAssistant;
+  private _config?: Config;
+
+  static get properties(): PropertyDeclarations {
+    return {
+      hass: {},
+      _config: {},
+    };
+  }
+
+  public getCardSize() {
+    return 1;
+  }
+
+  public setConfig(config) {
+    if (!config || !config.entity) {
+      throw new Error("Invalid card configuration");
+    }
+    if (!isValidEntityId(config.entity)) {
+      throw new Error("Invalid Entity");
+    }
+    this._config = { min: 0, max: 100, ...config };
+  }
+
+  protected render() {
+    if (!this._config) {
+      return html``;
+    }
+    const stateObj = this.hass!.states[this._config.entity];
 
-/*
- * @appliesMixin EventsMixin
- */
-class HuiGaugeCard extends EventsMixin(PolymerElement) {
-  static get template() {
+    return html`
+    ${this.renderStyle()}
+    <ha-card  @click="${this._handleClick}">
+    ${
+      !stateObj
+        ? html`<div class="not-found">Entity not available: ${
+            this._config.entity
+          }</div>`
+        : html`
+          <div class='container'>
+            <div class='gauge-a'></div>
+            <div class='gauge-b'></div>
+            <div class='gauge-c' id='gauge'></div>
+            <div class='gauge-data'>
+              <div id='percent'>${stateObj.state}
+              ${this._config!.unit_of_measurement ||
+                stateObj.attributes.unit_of_measurement ||
+                ""}
+              </div>
+              <div id='title'>${this._config.title}
+
+              </div>
+            </div>
+          </div>
+        `
+    }
+    </ha-card>
+    `;
+  }
+
+  protected shouldUpdate(changedProps) {
+    if (changedProps.get("hass")) {
```

Comment:
This is an excellent check and we should add this to every card and entity row . Note that you should compare `this._config` to old one too.

Reply:
CC @iantrich